### PR TITLE
Llvm36: Simplify trace management

### PR DESCRIPTION
**Pull request by vlj**
Reduce amount of trace data and use block analysis instead.
This should reduce memory consumption by a wide margin in most game, and improve performances.

### CODE REVIEW

File: rpcs3/Gui/SettingsDialog.cpp [hunk 1]
```diff
@@ -404,11 +404,11 @@ SettingsDialog::SettingsDialog(wxWindow *parent)
 		txt_dbg_range_min->GetValue().ToLong(&minllvmid);
 		txt_dbg_range_max->GetValue().ToLong(&maxllvmid);
 		Ini.LLVMExclusionRange.SetValue(chbox_core_llvm_exclud->GetValue());
-		Ini.LLVMMinId.SetValue(minllvmid);
-		Ini.LLVMMaxId.SetValue(maxllvmid);
+		Ini.LLVMMinId.SetValue((unsigned long)minllvmid);
+		Ini.LLVMMaxId.SetValue((unsigned long)maxllvmid);
```
Comment (danilaml) [inline rpcs3/Gui/SettingsDialog.cpp]: should probably be unsigned int.

File: rpcs3/Ini.h [hunk 2]
```diff
@@ -93,9 +95,9 @@ class Inis
 	// Core
 	IniEntry<u8> CPUDecoderMode;
 	IniEntry<bool> LLVMExclusionRange;
-	IniEntry<int> LLVMMinId;
-	IniEntry<int> LLVMMaxId;
-	IniEntry<int> LLVMThreshold;
+	IniEntry<unsigned int> LLVMMinId;
+	IniEntry<unsigned int> LLVMMaxId;
+	IniEntry<unsigned int> LLVMThreshold;
```
Comment (Nekotekina) [inline rpcs3/Ini.h]: u32 and other types were defined for what purpose?

Comment (danilaml) [inline rpcs3/Ini.h]: padhandler stuff is Int to though.

**Comment by vlj** [comment]
Replaced unsigned with u32, and removed more now useless trace related code.

**Comment by danilaml** [comment]
This increases "latency" a lot (time it takes to recompiler to speed up). Also for some reason it breaks sound in Sonic CD. It works at first but very garbled. And with very high chance (with ogl + vsync at least) it disappears in "press start" screen.

**Comment by vlj** [comment]
yes but there's no other way to do if we want to decrease memory usage in games.

**Comment by tambry** [comment]
@vlj I'd rather sacrifice speed/memory consumption for emulation accuracy (sound in this case), except when the accuracy improvement can't be heard or seen.

**Comment by vlj** [comment]
It has nothing to do with emulation accuracy. At worst it hints to a bug in audio implementation of rpcs3 since the audio thread seems to die when it looses some timing. 

Improvements? It makes llvm recompiler possible to debug. The subfic miscompilation was discovered thanks to this patch.

**Comment by tambry** [comment]
@vlj I knew that "accuracy" wasn't the right word for it, but couldn't find a better one.

**Comment by Nekotekina** [comment]
Merged.

**Pull request closed without merging**